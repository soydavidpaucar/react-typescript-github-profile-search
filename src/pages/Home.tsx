import { Alert, AlertIcon } from '@chakra-ui/react';
import SearchBar from '../components/SearchBar';
import useFetch from '../hooks/useFetch';

function Home() {
  const { githubData, fetchGithubData } = useFetch();

  return (
    <>
      <SearchBar onSearchGithubUsername={fetchGithubData} />
      {githubData.message && (
        <Alert status="error">
          <AlertIcon />
          Github user not found, please try again
        </Alert>
      )}
    </>
  );
}

export default Home;
