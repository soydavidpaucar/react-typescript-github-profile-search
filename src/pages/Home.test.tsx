import { describe, expect, it } from 'vitest';
import { act, fireEvent, getByLabelText, render } from '@testing-library/react';
import Home from './Home';

describe('Home', () => {
  it('should match with snapshot', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  it('should add username', async () => {
    const { getByRole } = render(<Home />);

    const input = getByRole('textbox') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'facebook' } });

    const button = getByRole('button') as HTMLButtonElement;

    await act(() => {
      fireEvent.click(button);
    });

    expect(input).toHaveValue('facebook');
  });

  it('should not add username when input is empty', async () => {
    const { getByRole } = render(<Home />);

    const input = getByRole('textbox') as HTMLInputElement;

    fireEvent.change(input, { target: { value: '' } });

    const button = getByRole('button') as HTMLButtonElement;

    await act(() => {
      fireEvent.click(button);
    });

    expect(input).toHaveValue('');
  });
});
