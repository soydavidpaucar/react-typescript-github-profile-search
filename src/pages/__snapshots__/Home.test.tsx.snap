// Vitest Snapshot v1

exports[`Home > should match with snapshot 1`] = `
<div>
  <section
    class="max-w-screen-md flex flex-col m-auto pt-52 px-12 sm:px-0"
  >
    <h1
      class="text-center text-[#ccd6f6] mb-12 text-4xl md:text-6xl"
    >
      Github Search Profile
    </h1>
    <form
      class="flex w-full mx-auto flex-col"
    >
      <div
        class="flex"
      >
        <input
          class="bg-transparent border px-4 py-2 text-[#ccd6f6] outline-none placeholder-[#ccd6f6] border-[#ccd6f6] rounded-md w-full text-sm md:text-base"
          name="username"
          placeholder="Enter github username..."
          type="text"
          value=""
        />
        <button
          class="text-[#64ffda] border-[#64ffda] border px-4 py-2 rounded-md ml-3 hover:bg-[#64ffda] hover:text-[#0a192f] transition text-sm md:text-base"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </section>
</div>
`;
